subprojects/file-watching/src/main/java/org/gradle/internal/watch/registry/impl/NonHierarchicalFileWatcherUpdater.java

```diff
@@ -103,7 +103,7 @@ public SnapshotHierarchy buildFinished(SnapshotHierarchy root) {
     private boolean containsSnapshots(File location, SnapshotHierarchy root) {
         CheckIfNonEmptySnapshotVisitor checkIfNonEmptySnapshotVisitor = new CheckIfNonEmptySnapshotVisitor(watchableHierarchies);
         root.visitSnapshotRoots(location.getAbsolutePath(), checkIfNonEmptySnapshotVisitor);
-        return checkIfNonEmptySnapshotVisitor.isEmpty();
+        return !checkIfNonEmptySnapshotVisitor.isEmpty();
```

Comment:
I was surprised by this change. How did this happen?

Reply:
I didn't have tests, yet. After adding the tests I found this error.

Reply:
I was just looking at the latest changes, and I assumed the code you fixed was there before the PR, and that's why I was wondering how it could have gone unnoticed. Now looking at all the changes I see that the fixed code was also introduced in this PR. So no worries.